task_stages.go

```diff
@@ -0,0 +1,114 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+// Compile-time proof of interface  implementation
+var _ TaskStages = (*taskStages)(nil)
+
+// TaskStages describes all the task stage related methods that the TFC/E API
+// supports.
+// **Note: This API is still in BETA and is subject to change.**
+type TaskStages interface {
+	// Read a task stage by ID
+	Read(ctx context.Context, taskStageID string, options *TaskStageReadOptions) (*TaskStage, error)
+
+	// List all task stages for a given rrun
+	List(ctx context.Context, runID string, options *TaskStageListOptions) (*TaskStageList, error)
+}
+
+// taskStages implements TaskStages
+type taskStages struct {
+	client *Client
+}
+
+// Stage is an enum that represents the possible run stages for run tasks
+type Stage string
+
+const (
+	PostPlan Stage = "post_plan"
+	PreApply Stage = "pre_apply"
+)
+
+// TaskStage represents a TFC/E run's stage where run tasks can occur
+type TaskStage struct {
+	ID               string                    `jsonapi:"primary,task-stages"`
+	Stage            Stage                     `jsonapi:"attr,stage"`
+	StatusTimestamps TaskStageStatusTimestamps `jsonapi:"attr,status-timestamps"`
+	CreatedAt        time.Time                 `jsonapi:"attr,created-at,iso8601"`
+	UpdatedAt        time.Time                 `jsonapi:"attr,updated-at,iso8601"`
+
+	Run         *Run          `jsonapi:"relation,run"`
+	TaskResults []*TaskResult `jsonapi:"relation,task-results"`
+}
+
+// TaskStageList represents a list of task stages
+type TaskStageList struct {
+	*Pagination
+	Items []*TaskStage
+}
+
+// TaskStageStatusTimestamps represents the set of timestamps recorded for a task stage
+type TaskStageStatusTimestamps struct {
+	ErroredAt  time.Time `jsonapi:"attr,errored-at,rfc3339"`
+	RunningAt  time.Time `jsonapi:"attr,running-at,rfc3339"`
+	CanceledAt time.Time `jsonapi:"attr,canceled-at,rfc3339"`
+	FailedAt   time.Time `jsonapi:"attr,failed-at,rfc3339"`
+	PassedAt   time.Time `jsonapi:"attr,passed-at,rfc3339"`
+}
+
+// TaskStageReadOptions represents the set of options when reading a task stage
+type TaskStageReadOptions struct {
+	Include string `url:"include"`
```

Comment:
We should make this a typed string out of the gate so we don't have to change it in 1.0